app/components/record_details/view.rb

```diff
@@ -5,14 +5,15 @@ class View < GovukComponent::Base
     include SanitizeHelper
     include SummaryHelper
 
-    attr_reader :trainee, :last_updated_event, :not_provided_copy, :show_provider, :show_record_source, :editable
+    attr_reader :trainee, :last_updated_event, :not_provided_copy, :show_provider, :show_record_source, :editable, :current_user
 
-    def initialize(trainee:, last_updated_event:, show_provider: false, show_record_source: false, editable: false)
+    def initialize(trainee:, last_updated_event:, show_provider: false, show_record_source: false, editable: false, current_user: nil)
       @trainee = trainee
       @last_updated_event = last_updated_event
       @show_provider = show_provider
       @editable = editable
       @show_record_source = show_record_source
+      @current_user = current_user
```

Comment:
given there's precedent in this params list for passing in a boolean for authorization (`@show_provider`), i suggest we roll with something similar here